stl/inc/algorithm

```diff
@@ -2348,6 +2348,83 @@ _FwdIt2 replace_copy_if(_ExPo&&, _FwdIt1 _First, _FwdIt1 _Last, _FwdIt2 _Dest, _
 }
 #endif // _HAS_CXX17
 
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // VARIABLE ranges::fill
+    class _Fill_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <class _Ty, output_iterator<const _Ty&> _It, sentinel_for<_It> _Se>
+        constexpr _It operator()(_It _First, _Se _Last, const _Ty& _Value) const {
+            _Adl_verify_range(_First, _Last);
+            auto _UFirst      = _Get_unwrapped(_STD move(_First));
+            const auto _ULast = _Get_unwrapped(_STD move(_Last));
+            if constexpr (_Fill_memset_is_safe<decltype(_UFirst), _Ty>) {
+#ifdef __cpp_lib_is_constant_evaluated
+                if (!_STD is_constant_evaluated())
+#endif // __cpp_lib_is_constant_evaluated
+                {
+                    _CSTD memset(_UFirst, static_cast<unsigned char>(_Value), static_cast<size_t>(_ULast - _UFirst));
+                    _UFirst += static_cast<size_t>(_ULast - _UFirst);
+                    _Seek_wrapped(_First, _UFirst);
+                    return _First;
+                }
+            }
+
+            while (_UFirst != _ULast) {
+                *_UFirst = _Value;
+                ++_UFirst;
+            }
+
+            _Seek_wrapped(_First, _STD move(_UFirst));
+            return _First;
+        }
+
+        template <class _Ty, output_range<const _Ty&> _Rng>
+        constexpr borrowed_iterator_t<_Rng> operator()(_Rng&& _Range, const _Ty& _Value) const {
+            return (*this)(_RANGES begin(_Range), _RANGES end(_Range), _Value);
+        }
+    };
+
+    inline constexpr _Fill_fn fill{_Not_quite_object::_Construct_tag{}};
+
+    // VARIABLE ranges::fill_n
+    class _Fill_n_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        template <class _Ty, output_iterator<const _Ty&> _It>
+        constexpr _It operator()(_It _First, iter_difference_t<_It> _Count, const _Ty& _Value) const {
+            if (0 < _Count) {

```

Comment:
Are we in favor of preserving [yoda conditions](https://en.wikipedia.org/wiki/Yoda_conditions)?

Reply:
I tend to think that they should be avoided, unless there's a compelling reason to consistently use `<` or something like that. Code usually reads more naturally when phrased as `if (unknown_thing op known_thing)`. Occasionally, something like `if (lo <= x && x <= hi)` is natural to write, which I don't think exactly counts as yoda.

One of the classic reasons to prefer yoda conditions was defending against `if (x = y)` but (1) we have compiler warnings for that and (2) `const` further defends against that when we can use it.

Reply:
Line 2413 says `_Count > 0` so this one should be consistent, good catch 😺

(This tends to vary between functions in the same file as we write new code, but functions should be internally consistent.)